fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java

```diff
@@ -1004,11 +1023,49 @@ public SlotDescriptor registerColumnRef(TableName tblName, String colName) throw
                                                 newTblName == null ? d.getTable().getName() : newTblName.toString());
         }
 
+        LOG.debug("register column ref table {}, colName {}, col {}", tblName, colName, col.toSql());
+        if (col.getType().isVariantType() || (subColNames != null && !subColNames.isEmpty())) {
+            if (!col.getType().isVariantType()) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_ILLEGAL_COLUMN_REFERENCE_ERROR,
+                        Joiner.on(".").join(tblName.getTbl(), colName));
+            }
+            if (subColNames == null) {
+                // Root
+                subColNames = new ArrayList<String>();
+            }
+            String key = d.getAlias() + "." + col.getName();
+            if (subColumnSlotRefMap.get(key) == null) {
+                subColumnSlotRefMap.put(key, Maps.newTreeMap(
+                    new Comparator<List<String>>() {
+                        public int compare(List<String> lst1, List<String> lst2) {
+                            String str1 = String.join(".", lst1);
+                            String str2 = String.join(".", lst2);
+                            return str1.compareTo(str2);
+                        }
+                    }));
+            }
+            SlotDescriptor result = subColumnSlotRefMap.get(key).get(subColNames);
+            if (result != null) {
+                // avoid duplicate slots
+                return result;
+            }
+            result = globalState.descTbl.addSlotDescriptor(d);
+            LOG.debug("register slot descriptor {}", result);
+            result.setSubColLables(subColNames);
+            result.setColumn(col);
+            if (!subColNames.isEmpty()) {
+                result.setMaterializedColumnName(col.getName() + "." + String.join(".", subColNames));
+            }
+            result.setIsMaterialized(true);
+            result.setIsNullable(col.isAllowNull());
+            subColumnSlotRefMap.get(key).put(subColNames, result);
+            return result;
+        }
+
         // Make column name case insensitive
         String key = d.getAlias() + "." + col.getName();
         SlotDescriptor result = slotRefMap.get(key);
         if (result != null) {
-            result.setMultiRef(true);
             return result;
         }
         result = globalState.descTbl.addSlotDescriptor(d);
```

Comment:
So, there is not SlotDescriptor for the top variant column self.

Reply:
the top variant column slot desc could be used when quering it like `select v from tbl`, otherwise slot for it's subcolumn should be enough